MediaBrowser.MediaEncoding/Attachments/AttachmentExtractor.cs

```diff
@@ -93,25 +93,59 @@ public async Task<(MediaAttachment Attachment, Stream Stream)> GetAttachment(Bas
 
             await semaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
 
-            try
+            if (!Directory.Exists(outputPath))
             {
-                if (!Directory.Exists(outputPath))
+                try
                 {
                     await ExtractAllAttachmentsInternal(
                         _mediaEncoder.GetInputArgument(inputFile, mediaSource),
                         outputPath,
+                        false,
                         cancellationToken).ConfigureAwait(false);
                 }
+                finally
```

Comment:
If the directory exists you're not releasing the lock

Reply:
Ah, sorry, I didn't pay attention while changing it.